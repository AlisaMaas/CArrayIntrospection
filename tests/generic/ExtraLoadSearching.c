/**
*
* Make sure we don't consider things to be null
* terminated if there's an extra load blocking our
* path.
*
**/

void find(char *arg) {
    for (int i = 0; arg[i]; i++){
        find(arg);
    }
}

void load(char **argv) {
    find(*argv);
}