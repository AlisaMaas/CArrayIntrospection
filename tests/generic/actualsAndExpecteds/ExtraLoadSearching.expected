Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'load':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Determine whether and how to annotate each function with the null-terminated annotation':
find with argument 0 should be annotated NULL_TERMINATED (2).
